### PR TITLE
junit: handle <testsuites> root element (#35)

### PR DESCRIPTION
**Pull request by bagage**
I added the TestSuitesModel class even if currently no fields are used in test reports generated.

### CODE REVIEW

**Comment by BogdanLivadariu** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java  1
         
```


See the complete overview on [Codacy](https://www.codacy.com/app/bogdan-livadariu/bootstraped-multi-test-results-report/pullRequest?prid=250153&bid=3336457)

File: junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
+package com.github.bogdanlivadariu.reporting.junit.xml.models;
+
+import java.util.List;
+import java.util.UUID;
+
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlAttribute;
+import javax.xml.bind.annotation.XmlElement;
+import javax.xml.bind.annotation.XmlElementWrapper;
```
Comment (BogdanLivadariu) [inline junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java]: Codacy Issue found: [Avoid unused imports such as 'javax.xml.bind.annotation.XmlElementWrapper'](https://www.codacy.com/app/bogdan-livadariu/bootstraped-multi-test-results-report/file/2685562127/issues/source?bid=3293946&fileBranchId=3336457#l10)

File: junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    private String name;
+
+    private String uniqueID;
+
+    private String overallStatus;
+
+    @XmlElement(name = "testsuite")
+    private List<TestSuiteModel> testsuite;
+
+    public void postProcess() {
```
Comment (BogdanLivadariu) [inline junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java]: Codacy Issue found: [The method postProcess() has an NPath complexity of 894](https://www.codacy.com/app/bogdan-livadariu/bootstraped-multi-test-results-report/file/2685562127/issues/source?bid=3293946&fileBranchId=3336457#l40)

File: junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+                if (updateTime) {
+                    time = Double.parseDouble(time ? time : "0") + Double.parseDouble(suite.getTime());
+                }
+                if (updateFailures) {
+                    if (suite->getOverallStatus() == Constants.FAILED) {
+                        failures = Integer.parseInt(failures ? failures : "0") + 1;
+                    }
+                }
+                if (updateErrors) {
+                    if (suite->getOverallStatus() == Constants.ERRORED) {
```
Comment (BogdanLivadariu) [inline junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java]: Codacy Issue found: [These nested if statements could be combined](https://www.codacy.com/app/bogdan-livadariu/bootstraped-multi-test-results-report/file/2685562127/issues/source?bid=3293946&fileBranchId=3336457#l57)

File: junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        Boolean updateTime = (time == null);
+        Boolean updateFailures = (failures == null);
+        Boolean updateErrors = (errors == null);
+        if (updateTime || updateFailures || updateErrors) {
+            for (TestSuiteModel suite : testsuite) {
+                if (updateTime) {
+                    time = Double.parseDouble(time ? time : "0") + Double.parseDouble(suite.getTime());
+                }
+                if (updateFailures) {
+                    if (suite->getOverallStatus() == Constants.FAILED) {
```
Comment (BogdanLivadariu) [inline junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java]: Codacy Issue found: [These nested if statements could be combined](https://www.codacy.com/app/bogdan-livadariu/bootstraped-multi-test-results-report/file/2685562127/issues/source?bid=3293946&fileBranchId=3336457#l52)

**Comment by BogdanLivadariu** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Clones added
============
- junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java  1
         
```


See the complete overview on [Codacy](https://www.codacy.com/app/bogdan-livadariu/bootstraped-multi-test-results-report/pullRequest?prid=250153&bid=3336457)

File: junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java (hunk 2 above)
Comment (BogdanLivadariu) [inline junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java]: Codacy Issue found: [The method postProcess() has an NPath complexity of 894](https://www.codacy.com/app/bogdan-livadariu/bootstraped-multi-test-results-report/file/2685562227/issues/source?bid=3293946&fileBranchId=3336457#l40)

File: junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java (hunk 1 above)
Comment (BogdanLivadariu) [inline junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java]: Codacy Issue found: [Avoid unused imports such as 'javax.xml.bind.annotation.XmlElementWrapper'](https://www.codacy.com/app/bogdan-livadariu/bootstraped-multi-test-results-report/file/2685562227/issues/source?bid=3293946&fileBranchId=3336457#l10)

File: junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java (hunk 3 above)
Comment (BogdanLivadariu) [inline junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java]: Codacy Issue found: [These nested if statements could be combined](https://www.codacy.com/app/bogdan-livadariu/bootstraped-multi-test-results-report/file/2685562227/issues/source?bid=3293946&fileBranchId=3336457#l57)

File: junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java (hunk 4 above)
Comment (BogdanLivadariu) [inline junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java]: Codacy Issue found: [These nested if statements could be combined](https://www.codacy.com/app/bogdan-livadariu/bootstraped-multi-test-results-report/file/2685562227/issues/source?bid=3293946&fileBranchId=3336457#l52)

File: junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/builder/JUnitReportBuilder.java [hunk 5]
```diff
@@ -44,6 +45,19 @@ public JUnitReportBuilder(List<String> xmlReports, String targetBuildPath) throw
         return processedTestSuites;
     }
 
+    private void processSuite(TestSuiteModel suite) {
```
Comment (BogdanLivadariu) [inline junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/builder/JUnitReportBuilder.java]: Codacy Issue found: [Avoid unused method parameters such as 'suite'.](https://www.codacy.com/app/bogdan-livadariu/bootstraped-multi-test-results-report/file/2685562220/issues/source?bid=3293946&fileBranchId=3336457#l48)

**Comment by BogdanLivadariu** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 5
           

Clones added
============
- junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java  1
         
```


See the complete overview on [Codacy](https://www.codacy.com/app/bogdan-livadariu/bootstraped-multi-test-results-report/pullRequest?prid=250153&bid=3336457)

File: junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/builder/JUnitReportBuilder.java (hunk 5 above)
Comment (BogdanLivadariu) [inline junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/builder/JUnitReportBuilder.java]: Codacy Issue found: [Avoid unused method parameters such as 'suite'.](https://www.codacy.com/app/bogdan-livadariu/bootstraped-multi-test-results-report/file/2685566576/issues/source?bid=3293946&fileBranchId=3336457#l48)

File: junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java (hunk 2 above)
Comment (BogdanLivadariu) [inline junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java]: Codacy Issue found: [The method postProcess() has an NPath complexity of 894](https://www.codacy.com/app/bogdan-livadariu/bootstraped-multi-test-results-report/file/2685566583/issues/source?bid=3293946&fileBranchId=3336457#l40)

File: junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+        Boolean updateTime = (time == null);
+        Boolean updateFailures = (failures == null);
+        Boolean updateErrors = (errors == null);
+        if (updateTime || updateFailures || updateErrors) {
+            for (TestSuiteModel suite : testsuite) {
+                if (updateTime) {
+                    time = Double.parseDouble(time ? time : "0") + Double.parseDouble(suite.getTime());
+                }
+                if (updateFailures) {
+                    if (suite.getOverallStatus() == Constants.FAILED) {
```
Comment (BogdanLivadariu) [inline junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java]: Codacy Issue found: [These nested if statements could be combined](https://www.codacy.com/app/bogdan-livadariu/bootstraped-multi-test-results-report/file/2685566583/issues/source?bid=3293946&fileBranchId=3336457#l52)

File: junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+                        failures = Integer.parseInt(failures ? failures : "0") + 1;
+                    }
+                }
+                if (updateErrors) {
+                    if (suite.getOverallStatus() == Constants.ERRORED) {
```
Comment (BogdanLivadariu) [inline junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java]: Codacy Issue found: [These nested if statements could be combined](https://www.codacy.com/app/bogdan-livadariu/bootstraped-multi-test-results-report/file/2685566583/issues/source?bid=3293946&fileBranchId=3336457#l57)

File: junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java (hunk 1 above)
Comment (BogdanLivadariu) [inline junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java]: Codacy Issue found: [Avoid unused imports such as 'javax.xml.bind.annotation.XmlElementWrapper'](https://www.codacy.com/app/bogdan-livadariu/bootstraped-multi-test-results-report/file/2685566583/issues/source?bid=3293946&fileBranchId=3336457#l10)

**Comment by BogdanLivadariu** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 5
           

Clones added
============
- junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java  1
         
```


See the complete overview on [Codacy](https://www.codacy.com/app/bogdan-livadariu/bootstraped-multi-test-results-report/pullRequest?prid=250153&bid=3336457)

File: junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+    private String name;
+
+    private String uniqueID;
+
+    private String overallStatus;
+
+    @XmlElement(name = "testsuite")
+    private List<TestSuiteModel> testsuite;
+
+    public void postProcess() {
```
Comment (BogdanLivadariu) [inline junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java]: Codacy Issue found: [The method postProcess() has an NPath complexity of 894](https://www.codacy.com/app/bogdan-livadariu/bootstraped-multi-test-results-report/file/2685602305/issues/source?bid=3293946&fileBranchId=3336457#l40)

File: junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java [hunk 9]
```diff
@@ -0,0 +1,106 @@
+package com.github.bogdanlivadariu.reporting.junit.xml.models;
+
+import java.util.List;
+import java.util.UUID;
+
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlAttribute;
+import javax.xml.bind.annotation.XmlElement;
+import javax.xml.bind.annotation.XmlElementWrapper;
```
Comment (BogdanLivadariu) [inline junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java]: Codacy Issue found: [Avoid unused imports such as 'javax.xml.bind.annotation.XmlElementWrapper'](https://www.codacy.com/app/bogdan-livadariu/bootstraped-multi-test-results-report/file/2685602305/issues/source?bid=3293946&fileBranchId=3336457#l10)

File: junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/builder/JUnitReportBuilder.java (hunk 5 above)
Comment (BogdanLivadariu) [inline junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/builder/JUnitReportBuilder.java]: Codacy Issue found: [Avoid unused method parameters such as 'suite'.](https://www.codacy.com/app/bogdan-livadariu/bootstraped-multi-test-results-report/file/2685602298/issues/source?bid=3293946&fileBranchId=3336457#l48)

**Comment by BogdanLivadariu** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Clones added
============
- junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java  1
         
```


See the complete overview on [Codacy](https://www.codacy.com/app/bogdan-livadariu/bootstraped-multi-test-results-report/pullRequest?prid=250153&bid=3336457)

File: junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+    private String name;
+
+    private String uniqueID;
+
+    private String overallStatus;
+
+    @XmlElement(name = "testsuite")
+    private List<TestSuiteModel> testsuite;
+
+    public void postProcess() {
```
Comment (BogdanLivadariu) [inline junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java]: Codacy Issue found: [The method postProcess() has an NPath complexity of 894](https://www.codacy.com/app/bogdan-livadariu/bootstraped-multi-test-results-report/file/2685730506/issues/source?bid=3293946&fileBranchId=3336457#l40)

File: junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java [hunk 11]
```diff
@@ -0,0 +1,102 @@
+package com.github.bogdanlivadariu.reporting.junit.xml.models;
+
+import java.util.List;
+import java.util.UUID;
+
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlAttribute;
+import javax.xml.bind.annotation.XmlElement;
+import javax.xml.bind.annotation.XmlElementWrapper;
```
Comment (BogdanLivadariu) [inline junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java]: Codacy Issue found: [Avoid unused imports such as 'javax.xml.bind.annotation.XmlElementWrapper'](https://www.codacy.com/app/bogdan-livadariu/bootstraped-multi-test-results-report/file/2685730506/issues/source?bid=3293946&fileBranchId=3336457#l10)

**Comment by BogdanLivadariu** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Clones added
============
- junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java  1
         
```


See the complete overview on [Codacy](https://www.codacy.com/app/bogdan-livadariu/bootstraped-multi-test-results-report/pullRequest?prid=250153&bid=3336457)

File: junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java (hunk 10 above)
Comment (BogdanLivadariu) [inline junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java]: Codacy Issue found: [The method postProcess() has an NPath complexity of 894](https://www.codacy.com/app/bogdan-livadariu/bootstraped-multi-test-results-report/file/2685750173/issues/source?bid=3293946&fileBranchId=3336457#l40)

File: junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java (hunk 11 above)
Comment (BogdanLivadariu) [inline junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java]: Codacy Issue found: [Avoid unused imports such as 'javax.xml.bind.annotation.XmlElementWrapper'](https://www.codacy.com/app/bogdan-livadariu/bootstraped-multi-test-results-report/file/2685750173/issues/source?bid=3293946&fileBranchId=3336457#l10)

**Comment by BogdanLivadariu** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Clones added
============
- junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java  1
         
```


See the complete overview on [Codacy](https://www.codacy.com/app/bogdan-livadariu/bootstraped-multi-test-results-report/pullRequest?prid=250153&bid=3336457)

File: junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+    private String name;
+
+    private String uniqueID;
+
+    private String overallStatus;
+
+    @XmlElement(name = "testsuite")
+    private List<TestSuiteModel> testsuite;
+
+    public void postProcess() {
```
Comment (BogdanLivadariu) [inline junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java]: Codacy Issue found: [The method postProcess() has an NPath complexity of 894](https://www.codacy.com/app/bogdan-livadariu/bootstraped-multi-test-results-report/file/2685956029/issues/source?bid=3293946&fileBranchId=3336457#l39)

**Comment by BogdanLivadariu** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Clones added
============
- junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java  1
         
```


See the complete overview on [Codacy](https://www.codacy.com/app/bogdan-livadariu/bootstraped-multi-test-results-report/pullRequest?prid=250153&bid=3336457)

File: junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java (hunk 12 above)
Comment (BogdanLivadariu) [inline junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java]: Codacy Issue found: [The method postProcess() has an NPath complexity of 894](https://www.codacy.com/app/bogdan-livadariu/bootstraped-multi-test-results-report/file/2695613449/issues/source?bid=3293946&fileBranchId=3336457#l39)

File: junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/builder/JUnitReportBuilder.java [hunk 13]
```diff
@@ -3,6 +3,7 @@
 import java.io.File;
 import java.io.FileNotFoundException;
 import java.io.IOException;
+import java.lang.ClassCastException;
```
Comment (BogdanLivadariu) [inline junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/builder/JUnitReportBuilder.java]: Codacy Issue found: [Avoid importing anything from the package java.lang](https://www.codacy.com/app/bogdan-livadariu/bootstraped-multi-test-results-report/file/2695613442/issues/source?bid=3293946&fileBranchId=3336457#l6)

**Comment by BogdanLivadariu** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Clones added
============
- junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java  1
         
```


See the complete overview on [Codacy](https://www.codacy.com/app/bogdan-livadariu/bootstraped-multi-test-results-report/pullRequest?prid=250153&bid=3336457)

**Comment by BogdanLivadariu** [comment]
@bagage maybe you can extract a base class from 
```
...dariu/reporting/junit/xml/models/TestSuitesModel.java
...dariu/reporting/junit/xml/models/TestSuitesModel.java
```

and extend it to fit our needs, thus avoiding code duplication

**Comment by BogdanLivadariu** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java  1
         
```


See the complete overview on [Codacy](https://www.codacy.com/app/bogdan-livadariu/bootstraped-multi-test-results-report/pullRequest?prid=250153&bid=3336457)

File: junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/builder/JUnitReportBuilder.java [hunk 14]
```diff
@@ -44,25 +44,37 @@ public JUnitReportBuilder(List<String> xmlReports, String targetBuildPath) throw
...
     }
 
-    private List<TestSuiteModel> processXmlReports(List<String> xmlReports) throws JAXBException {
-        JAXBContext cntx = JAXBContext.newInstance(TestSuiteModel.class);
-
-        Unmarshaller unm = cntx.createUnmarshaller();
+    private void processSuite(TestSuiteModel ts) {
+        for (TestCaseModel tc : ts.getTestcase()) {
+            if (tc.getError() != null) {
+                System.out.println();
```
Comment (BogdanLivadariu) [inline junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/builder/JUnitReportBuilder.java]: do we need this check ?

Comment (bagage) [inline junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/builder/JUnitReportBuilder.java]: I've just moved some code from one line to another; it was already here. But it looks unneeded in any case indeed.

File: junit-reporting-handlebars/src/test/resources/valid-report-2.xml [hunk 15]
```diff
@@ -0,0 +1,307 @@
...
+
+            <testcase name="verifyingTheClonedFile" time="0.068" classname="test.java.PlayingWithExcelFiles" />
+                <testcase name="verifyingTheClonedFile" time="0.065" classname="test.java.PlayingWithExcelFiles" />
+                <testcase name="verifyingTheClonedFile" time="0.044" classname="test.java.PlayingWithExcelFiles" />
+                <testcase name="verifyingTheClonedFile" time="0.120" classname="test.java.PlayingWithExcelFiles" />
+                <testcase name="verifyingTheClonedFile" time="0.051" classname="test.java.PlayingWithExcelFiles" />
+                <testcase name="verifyingTheClonedFile" time="0.048" classname="test.java.PlayingWithExcelFiles" />
+
+</testsuite>
+</testsuites>
```
Comment (BogdanLivadariu) [inline junit-reporting-handlebars/src/test/resources/valid-report-2.xml]: can you reindent the xml file ?

http://www.freeformatter.com/xml-formatter.html

**Comment by bagage** [comment]
@BogdanLivadariu I am not sure how/why both `TestSuiteModel` and `TestSuitesModel` should be factorized; they have nothing in common in practice except the `name` attribute. Actually I would love to not add `TestSuitesModel` since I do not use its attributes, but it seems necessary to unmarshall XML files.

File: junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/builder/JUnitReportBuilder.java (hunk 14 above)
Comment (BogdanLivadariu) [inline junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/builder/JUnitReportBuilder.java]: my bad then :) please remove it

File: junit-reporting-handlebars/src/test/resources/valid-report-2.xml (hunk 15 above)
Comment (bagage) [inline junit-reporting-handlebars/src/test/resources/valid-report-2.xml]: Again this is a copy&paste from valid-report-1.xml. Should I reformat both of them then?

**Comment by BogdanLivadariu** [comment]
just looked quick over codacy, and looks we cannot rely on it all the time :)

leave the classes as they are, we'll see if we need to refactor them after

**Comment by BogdanLivadariu** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java  1
         
```


See the complete overview on [Codacy](https://www.codacy.com/app/bogdan-livadariu/bootstraped-multi-test-results-report/pullRequest?prid=250153&bid=3336457)

File: junit-reporting-handlebars/src/test/resources/valid-report-2.xml (hunk 15 above)
Comment (BogdanLivadariu) [inline junit-reporting-handlebars/src/test/resources/valid-report-2.xml]: its ok to have all the xml files formatted, not sure why that one got away

**Comment by BogdanLivadariu** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- junit-reporting-handlebars/src/main/java/com/github/bogdanlivadariu/reporting/junit/xml/models/TestSuitesModel.java  1
         
```


See the complete overview on [Codacy](https://www.codacy.com/app/bogdan-livadariu/bootstraped-multi-test-results-report/pullRequest?prid=250153&bid=3336457)

**Comment by bagage** [comment]
OK, this time I am done! You can merge it if you wish and I'll gladly try it :).

**Comment by BogdanLivadariu** [comment]
great work @bagage will try to release a new version later today or tomorrow

**Pull request merged**